docs/container-vm-image-wrapper.md

```diff
@@ -0,0 +1,238 @@
+# VM Images in Container Registery
+
+## Motivation
+
+KubeVirt is a drop in solution for running VMs on an existing kubernetes
+cluster. One of our goals in doing this is to leverage the tools and workflows
+kubernetes has established for containers and apply them to virtualized
+workloads.
+
+For example, with KubeVirt VMs are managed like pods. Users post VMs to
+kubernetes like pods and the VMs will soon have the option to integrate into
+the kubernetes overlay network like pods.
+
+To continue down the path of allowing users to introduce VM workloads into
+an existing kubernetes cluster with as little friction as possible, the next
+hurdle for us to overcome is tieing VM image distribution into the same
+mechanism used by pods. To do this, KubeVirt must have a way of storing and
+accessing VM images using a container registry.
+
+## Requirements
+* Users must have a workflow for pushing VM images into a container registry.
+* KubeVirt runtime must be able to consume VM images stored in the container
+  registry.
+
+## Initial Use Cases
+For the sake of simplicity, this proposal is focusing on the details involved
+with use cases where VMs are launched using ephemeral storage that is not
+persistent between VM restarts.
```

Comment:
> IOW, I think the persistent vs ephemeral disk distinction ought to be independent of the how we get content into the volumes that back the disk.

Forget the persistent vs ephemeral disk distinction for now. We want people to look at KubeVirt and think "oh wow, so I just drop in KubeVirt runtime into my k8s cluster and push my VM images to the registery... then everything works. VMs workloads are just like containers for me, This is awesome"

To get that reaction we have to think backwards. It's not "how do we get KubeVirt to match the VM way of doing things", it's "how do we get KubeVirt to make the VM way of doing things match the container way of doing things"

I believe our success (on a mass adoption level) lies in being able to converge these workflows into one... *NOTE: I want to be careful I'm not misunderstood here though. I know KubeVirt will absolutely be supporting VM use cases that do not map directly to container workflows. I don't want to discuss those in this proposal though.*

In a similar way to how the CNI networking proposal was focused on how to tie VMs into the pod network, this proposal is focused on how to tie VMs into the way containers are built, distributed, and executed in the cluster.